src/sdam/topology.ts

```diff
@@ -469,7 +469,8 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
       selectServerOptions,
       (err, server) => {
         if (err) {
-          return this.close({ force: false }, () => exitWithError(err));
+          this.close();
```

Comment:
same question about errors

Reply:
|method|relevant direct dependencies|throws?|
|------|------------|-------|
|`Connection.destroy`| - | No |
|`ConnectionPool.close`| `Connection.destroy` | No |
|`Server.destroy` | `stateTransition`\*, `ConnectionPool.close` | No; `stateTransition`\* can technically throw, but `Server.destroy` is written such that it will never try to make an illegal state transition.|
|`destroyServer` | `Server.destroy` | No |
|`Topology.close` | `destroyServer`, `stateTransition`\*\* | No; `stateTransition`\*\* can technically throw, but like in `Server.destroy`, `Topology.close` is written such that it never tries to make an illegal state transition|

Note:

`stateTransition`\* refers to the local function using the static state machine defined for the `Server` class created by the `makeStateMachine` helper.

`stateTransition`\*\* refers to the local function using the static state machine defined for the `Topology` class created by the `makeStateMachine` helper.